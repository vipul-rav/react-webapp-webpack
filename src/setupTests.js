import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

let store;
beforeEach(() => {
  const mockStore = configureMockStore();
  store = mockStore();
});

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
