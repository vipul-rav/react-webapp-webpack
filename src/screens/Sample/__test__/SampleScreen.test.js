import React from 'react';
import { fireEvent, cleanup } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import { render } from '../../../setupTests';
import { SampleScreen } from '../SampleScreen';

const content = {
  ExitText: 'Exit',
  btnContinue: 'Continue',
  homePageText: 'Home Page'
};

const setup = (overridesProps) => {
  const props = {
    content: content,
    navigateToNextScreen: jest.fn(),
    getPostList: jest.fn(),
    ...overridesProps
  };

  const mockStore = configureMockStore();
  const store = mockStore({
    config: {
      externalContent: content
    },
    post: {
      postList: [
        {
          id: '1',
          title: 'title1'
        }
      ]
    }
  });

  const { container, getByText } = render(<SampleScreen {...props} store={store} />);

  return {
    props,
    container,
    getByText
  };
};

describe('Sample container', () => {
  afterEach(cleanup);
  it('should render', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
  it('has Continue button', () => {
    const { getByText } = setup();
    expect(getByText('Continue')).toBeInTheDocument();
  });
  it('click Continue button', () => {
    const { getByText, props } = setup();

    fireEvent.click(getByText('Continue'));
    expect(props.navigateToNextScreen).toBeCalled();
  });
});
