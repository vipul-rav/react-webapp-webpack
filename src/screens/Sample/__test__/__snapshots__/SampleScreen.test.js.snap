// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample container should render 1`] = `
<div>
  <div>
    <h1>
      Home Page
    </h1>
    <div>
      <ul />
    </div>
    <button
      class="c-a-button c-a-button--size-md c-a-button--primary"
      id="btnContinue"
    >
      <span
        class="c-a-button__title"
      >
        Continue
      </span>
    </button>
  </div>
</div>
`;
